akka-cluster-typed/src/main/scala/akka/cluster/typed/internal/receptionist/ClusterReceptionist.scala

```diff
@@ -160,8 +169,19 @@ private[typed] object ClusterReceptionist extends ReceptionistBehaviorProvider {
                   ServiceRegistry(registry).removeAll(removalForKey).toORMultiMap
                 }
             }
+
```

Comment:
There is something that doesn't match what I expected. We already had the removal retries for removed nodes, see `case RemoveTick`. The remaining problem was 

> when a registered actor is terminated, but the node is still up and running

We are watching registered actors, see `case ReceptionistMessages.Register`. It's the removal when such actors are terminated that can be lost. See `case RegisteredActorTerminated` so I expected the addition to tombstones there.

Registered actors are always supposed to be local actors. Perhaps we should add a check to prevent misusage.

Reply:
I think the lost removal could actually happen in two scenarios - 
1. the first one is the local watch + termination at the same time as the service key has modifications happen on another node and that is what the test case covers (actor1 stops triggering deathwatch). 
2. the node is removed and cleanup removes those actors, and there is a concurrent registration on another node so the removed actors are re-added and then the current logic will remove them again on next tick

However, I think that it would be good to not send an updated listing saying those actors are alive again, when we in fact know they are not, and to do that we'd need to tombstone here as well. (See line 251)

Reply:
This PR is missing 1 as far as I can see.

For 2 you would have to add all actors from the removed nodes to the tombsones, making it a rather big collection, and for those the tombstones are not needed for retry of the removals. You want them for the notifications, but that can be solved by checking if the address of the ActorRef is still in the cluster.members before sending out the notification.

Reply:
case 1 is covered, missed it at first

Reply:
Alright, sounds like that would cover it. I'll update accordingly.